sympy/core/function.py

```diff
@@ -1325,6 +1326,10 @@ def __new__(cls, expr, *variables, **kwargs):
         # we return here if evaluate is False or if there is no
         # _eval_derivative method
         if not evaluate or not hasattr(expr, '_eval_derivative'):
+            # return the derivative of Equality
+            if isinstance(expr, Equality):
```

Comment:
We do not have an Equation class, just `Eq` and an `Eq` is a boolean. And although I think there is some code somewhere (present, or removed) that tried to implement this, differentiating "both sides" is incompatible with the current definition of "Eq".